api/v1/health/openapi.yaml

```diff
@@ -0,0 +1,178 @@
+---
+swagger: '2.0'
+info:
+  title: Cilium-Health API
+  description: Cilium Health Checker
+  version: v1beta
+x-schemes:
+- unix
+basePath: "/v1beta"
+produces:
+- application/json
+consumes:
+- application/json
+paths:
+  "/hello":
+    get:
+      summary: Say hello to cilium-health
+      description: |
+        Returns a successful status code if this cilium-health instance is
+        reachable.
+      responses:
+        '200':
+          description: Success
+  "/healthz":
+    get:
+      summary: Get health of Cilium node
+      description: |
+        Returns health and status information of the local node including
+        load and uptime, as well as the status of related components including
+        the Cilium daemon.
+      responses:
+        '200':
+          description: Success
+          schema:
+            "$ref": "#/definitions/HealthResponse"
+        '500':
+          description: Failed to contact local Cilium daemon
+          x-go-name: Failed
+          schema:
+            "$ref": "../openapi.yaml#/definitions/Error"
+  "/status/probe":
+    put:
+      summary: Get connectivity status of the Cilium cluster
+      description: |
+        Runs a synchronous probe to all other cilium-health instances and
+        returns the connectivity status.
+      tags:
+      - connectivity
+      responses:
+        '200':
+          description: Success
+          schema:
+            "$ref": "#/definitions/StatusResponse"
+        '500':
+          description: Internal error occurred while conducting connectivity probe
+          x-go-name: Failed
+          schema:
+            "$ref": "../openapi.yaml#/definitions/Error"
+  "/status":
+    get:
+      summary: Get connectivity status of the Cilium cluster
+      description: |
+        Returns the connectivity status to all other cilium-health instances
+        using interval-based probing.
+      tags:
+      - connectivity
+      responses:
+        '200':
+          description: Success
+          schema:
+            "$ref": "#/definitions/StatusResponse"
+definitions:
+  HealthResponse:
+    description: Health and status information of local node
+    type: object
+    properties:
+      cilium:
+        description: Status of Cilium daemon
+        "$ref": "../openapi.yaml#/definitions/StatusResponse"
```

Comment:
So we have a `StatusResponse` in both yaml files? Can we rename one of them?

Reply:
Sure.